md/bdm-access-control.md

```diff
@@ -314,3 +314,153 @@ Deploy your access control file.
+It allows to protect the bdm instances by limiting the access to the jpql requests that retrieve the object instances rather than to the instances themselves.
+:::
+
+**Scenario:**
+
+We will look into a mark contestation process for a school. Students can make requests about their marks to their teachers through a BPM process.
+Each teacher teaches a different subject. A teacher can only access requests made concerning his subject.
+
+The step by step guide assumes a working Bonita Studio v. 7.7.x +, with the Acme organization deployed.
+
+**Step by step implementation:**
+
+**1. Define the BDM**
+
+We will use a simple relatively simple BDM for this use case: 
+
+1. an object Student
+
+| Name | Type |
+|---|---|
+| fullname | STRING |
+
+2. an object Request
+
+ | Name | Type |
+|---|---|
+| subject | STRING |
+| medicalComment | STRING |
+| content | STRING |
+| student | Student |
+
+Subject and Student are mandatory fields.
+We also define a custom query on the Request object, *findBySubject*:
+
+```
+SELECT r
+FROM Request r
+WHERE r.subject= :subject
+ORDER BY r.persistenceId
+
+```
+
+**2. Define the profiles**
+
+We will need a few new profiles, mapped to one or more users. We need one profile per subject (), as well as a **Teacher** profile (for all the teachers). A maths teacher would therefore have **MathematicsTeacher** and a **Teacher** profile. In the rest of the guide we will use 3 profiles **MathematicsTeacher**, **PhysicsTeacher** and **Teacher**, mapped to **helen.kelly**,**jan.fischer** and both respectively.
+For the details on how to easily create and map those profiles see the [Profile Creation](profileCreation.md) page.
+
+**3. Create a process and generate some request instances**
+
+You will need some instances of the request object, as well as some students. To create them, follow the steps discribed in the section  **II - Invoice**, part 3 *Create a process and generate some data* of this very page.
+
+**4. Create the visualisation application**
+
+We will now create an application for the teachers to review the student requests : 
+
+In the UI Designer, create an application page (*reviewRequests*):
+
+- Create 2 new variables
+    - **Name** : requestList
+    - **Type** : External API
+    - **API URL**: ../API/bdm/businessData/com.company.model.Request?q=findBySubject&p=0&c=10&f=subject%3D{{selectedSubject}}
+    
+    -  **Name** : selectedSubject
+	-  **Type** : String
+    
+- Add a new Select box to the page (To choose beetween subjects):
+    - **Label** : Subject class
+    - **Available Values** : Mathematics, Physics (constants).
+    - **Value** : selectedSubject
+    
+- Add  a new Table widget to the page (To display the requests):
+    - **Headers** : Id, Subject, Content, Medical comment, Student (constants)
+    - **Content** : requestList (script)
```

Comment:
**Content** : requestList (script, click the X icon to switch from contstant to script (bound data?))

Reply:
~X~ => **fx**